sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -1269,6 +1274,50 @@ private void applyBulkVersioning(Class<?> domainType,
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private <T, S extends T> PartitionKey getPartitionKeyFromValue(CosmosEntityInformation<T, ?> information, S entity) {
+        Object pkFieldValue = information.getPartitionKeyFieldValue(entity);
+        PartitionKeyBuilder partitionKeyBuilder = new PartitionKeyBuilder();
+        PartitionKey partitionKey;
+        if (pkFieldValue.getClass().isArray() || pkFieldValue instanceof Collection<?>) {
+            for (final Object pkValue : (ArrayList<Object>) pkFieldValue) {
```

Comment:
Probably can reusing` PartitionKeyInternal fromObjectArray(Object[] values, boolean strict)` from the core SDK

Reply:
@xinlian12 PartitionKeyInternal is not available in the Spring SDK

Reply:
I would rather ad a public helper method on PartitionKeyBuilder
To construct from value - opening implementation package to azure-spring-data-cosmos seems to open too wide of a surface area.

Reply:
I think this is a good point, we can allow the PartitionKeyBuilder to accept a list of objects, and then internally using the PartitionKeyInternal fromObjectArray

Reply:
Done